src/File.re

```diff
@@ -1,4 +1,17 @@
+open LidcoreBsNode;
+
 type t;
 
+[@bs.send] external createReadStream : t => LidcoreBsNode.Stream.readable = "";
+
 [@bs.send]
-external exists : (t, (Js.nullable(exn), bool) => unit) => unit = "";
+external createWriteStream : t => LidcoreBsNode.Stream.writable = "";
+
+[@bs.send] external exists : (t, Callback.callback(bool)) => unit = "";
+
+let download = (f, path) => {
+  let readable = createReadStream(f);
+  let writeable = Fs.createWriteStream(~path, ());
+  Stream.pipe(readable, writeable);
+  Stream.on(writeable, `Finish(() => Js.log("finished")));
+};
```

Comment:
Yup this looks good! If you'd wanted to make `download ` a `Callback.t unit` computation, you'd do:
```
let download = (f, path, cb) => {
  let readable = createReadStream(f);
  let writeable = Fs.createWriteStream(~path, ());
  Stream.pipe(readable, writeable);
  Stream.on(writeable, `Finish(() => Callback.return () cb));
  Stream.on(writeable, `Error(exn => Callback.fail exn cb));
};
```

Reply:
Which can then be composed:
```
(* Sorry OCaml syntax.. *)
let download_if_exists =
  let f = Bucket.file (...) in
  File.exists f >> fun (exists) ->
    if exists then
      download f >> fun () ->
        Callback.return true
    else
      Callback.return false
> download_if_exists : bool Callback.t
```